test/extensions/filters/http/lua/lua_integration_test.cc

```diff
@@ -974,6 +992,47 @@ name: lua
   testRewriteResponse(FILTER_AND_CODE);
 }
 
+// Rewrite response buffer, without original upstream response body
+// and always warp body.
+TEST_P(LuaIntegrationTest, RewriteResponseBufferWithoutUpstreamBody) {
+  const std::string FILTER_AND_CODE =
+      R"EOF(
+name: lua
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
+  inline_code: |
+    function envoy_on_response(response_handle)
+      local content_length = response_handle:body(true):setBytes("ok")
+      response_handle:logTrace(content_length)
+
+      response_handle:headers():replace("content-length", content_length)
+    end
+)EOF";
+
+  testRewriteResponseWithoutUpstreamBody(FILTER_AND_CODE, true);
+}
+
+// Rewrite response buffer, without original upstream response body
+// and don't always wrap body.
+TEST_P(LuaIntegrationTest, RewriteResponseBufferWithoutUpstreamBodyAndDisableWrapBody) {
+  const std::string FILTER_AND_CODE =
+      R"EOF(
+name: lua
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
+  inline_code: |
+    function envoy_on_response(response_handle)
+      if response_handle:body(false) then
+        local content_length = response_handle:body():setBytes("ok")
```

Comment:
I'm confused. Shouldn't this end up returning nil and the script should crash? Shouldn't we be verifying nil here instead?

Reply:
The `response_handle:body(false)` always returns nil so this branch is skipped. It is just a mirror of the previous test but with the `false` given.

Reply:
Ah sorry, I missed that. OK LGTM modulo the typo.